linux/hid.c

```diff
@@ -790,6 +791,25 @@ int HID_API_EXPORT hid_get_feature_report(hid_device *dev, unsigned char *data,
 	return res;
 }
 
+int HID_API_EXPORT HID_API_CALL hid_get_input_report(hid_device *dev, unsigned char *data, size_t length)
+{
+	int res;
+	unsigned char bfr[64] = {0}; //Maximum Hid Report Size is 64 bytes, added one extra. See https://www.usb.org/sites/default/files/documents/hid1_11.pdf section 5.6
+
+	if(length > 64){
+		register_device_error_format(dev, "ioctl (GREPORT): %s", strerror(EINVAL)); //22      /* Invalid argument */
+	}
+
+	memcpy(bfr, data, length); // copy the data onto the buffer	
+
+	res = ioctl(dev->device_handle, HIDIOCGREPORT, bfr);
```

Comment:
`HIDIOCGREPORT` [expects a `hiddev_report_info`](https://elixir.bootlin.com/linux/v4.4/source/include/uapi/linux/hiddev.h#L160), which is [declared here](https://elixir.bootlin.com/linux/v4.4/source/include/uapi/linux/hiddev.h#L65).

This differs from `HIDIOCGFEATURE` which takes in the [first byte as a report ID](https://elixir.bootlin.com/linux/v4.4/source/include/uapi/linux/hidraw.h#L40).

I'm also not sure that this is a valid `ioctl` for `hidraw` -- this looks like  it is for `hiddev`!

Reply:
also, I don't see support of Input Reports by hidraw at all

Reply:
Nope, [it doesn't](https://elixir.bootlin.com/linux/v5.3-rc3/source/drivers/hid/hidraw.c#L424)